cra-device-class-aware-loading/server.js

```diff
@@ -0,0 +1,109 @@
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const path = require('path');
+const cors = require('cors');
+const app = express();
+const DeviceApiWeb = require('deviceatlas-deviceapi').DeviceApiWeb;
+const stringSimilarity = require('string-similarity');
+
+app.disable('x-powered-by');
+app.use(cors());
+app.use(express.static(path.join(__dirname, 'build')));
+
+// DeviceAtlas server-side API
+const deviceApi = (function () {
+  const devApi = new DeviceApiWeb();
+  
+  try {
+    devApi.loadDataFromFile('./lib/device-atlas/device-data-file/55268_20190722.json');
+  } catch (ex) {
+    devApi.error = ex;
+    console.log(ex.message);
+  }
+  
+  return devApi;
+})();
+
+// TODO: device properties testing -> remove on release
+// app.get('/api/device-properties', function (req, res) {
+//   if (deviceApi.error) {
+//     return res.status(500).json({
+//       message: deviceApi.error
+//     });
+//   }
+//   const properties = deviceApi.getPropertiesFromRequest(req);
+//   const props = {};
+//   for (const name in properties.getMap()) {
+//       props[name] = {};
+//       props[name]['value'] = properties.get(name).getValue();
+//   }
+//   return res.status(200).send(props);
+// });
+
+app.get('/api/device', function (req, res) {
+  console.log('[server] user-agent => ', req.headers['user-agent']);
+  // TODO: device properties testing -> remove on release
+  // const similarity1 = stringSimilarity.compareTwoStrings('Galaxy Tab 4 10.1 LTE', 'Galaxy Tab 4 10.1');
+  // const similarity2 = stringSimilarity.compareTwoStrings('Motorola Moto X 2nd Gen', 'Motorola Moto X');
+  // console.log('similarity1, similarity2 => ', similarity1, similarity2);
+  // similarity1, similarity2 =>  0.896551724137931 0.8
+
+  if (deviceApi.error) {
+    return res.status(500).json({
+      message: deviceApi.error
+    });
+  }
+  
+  const properties = deviceApi.getPropertiesFromRequest(req);
+
+  const isMobileDevice = properties.get('mobileDevice').getValue();
+  if (!isMobileDevice) {
+    return res.status(400).json({
+      message: 'Currently only support Android/iOS devices.'
```

Comment:
`We currently only support detecting Android and iOS devices`.

Reply:
Yes, replaced.